product_pricelist_direct_print_company_group/wizards/product_pricelist_print.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2022 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, models
+
+
+class ProductPricelistPrintCompanyGroup(models.TransientModel):
+    _inherit = "product.pricelist.print"
+
+    @api.model
+    def _get_sale_order_domain(self, partner):
+        domain = super()._get_sale_order_domain(partner)
+        if partner.company_group_member_ids:
+            domain.remove(("partner_id", "child_of", partner.id))
+            domain.append(("company_group_id", "=", partner.id))
```

Comment:
```suggestion
            domain.append(("partner_id", "in", partner.company_group_member_ids.ids))
```
With this change we exploit the partner field index:
https://github.com/odoo/odoo/blob/739da6b8b2ba60965728ffdc1ba133203fd65a2d/addons/sale/models/sale.py#L190
company_group_id field are not indexed:
https://github.com/OCA/partner-contact/blob/8f615c685205bb44ea785f5e3e64efd54ce5efae/partner_company_group/models/sale_order.py#L11